pkg/sessions/redis/redis_store.go

```diff
@@ -287,6 +245,77 @@ func (store *SessionStore) storeValue(ctx context.Context, value string, expirat
 	return ticket.encodeTicket(store.CookieOptions.Name), nil
 }
 
+// loadSessionFromTicket loads the session based on the ticket value
+func (store *SessionStore) loadSessionFromTicket(ctx context.Context, value string) (*sessions.SessionState, error) {
+	ticket, err := decodeTicket(store.CookieOptions.Name, value)
+	if err != nil {
+		return nil, err
+	}
+
+	resultBytes, err := store.Client.Get(ctx, ticket.asHandle(store.CookieOptions.Name))
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := encryption.NewGCMCipher(ticket.Secret)
+	if err != nil {
+		return nil, err
+	}
+
+	session, err := sessions.DecodeSessionState(resultBytes, c, false)
+	if err != nil {
+		// The GCM cipher will error due to a legacy JSON payload not passing
+		// the authentication check part of AES GCM encryption.
+		// In that case, we can attempt to fallback to try a legacy load
+		return store.legacyV5LoadSessionFromTicket(ctx, value)
+	}
+	return session, nil
+}
+
+// legacyV5LoadSessionFromTicket loads the session based on the ticket value
+// This falls uses V5 style encryption of Base64 + AES CFB
+func (store *SessionStore) legacyV5LoadSessionFromTicket(ctx context.Context, value string) (*sessions.SessionState, error) {
```

Comment:
Hmmm, I think I need to circle back on this one later.  I don't think I have test coverage for this legacy fallback.